### PR TITLE
fix float to i128 cast

### PR DESCRIPTION
**Pull request by houqp**
caught by datafusion integration test, see: https://github.com/apache/arrow-datafusion/pull/1733

### CODE REVIEW

File: src/compute/cast/primitive_to.rs [hunk 1]
```diff
@@ -237,10 +235,11 @@ pub(super) fn float_to_decimal_dyn<T>(
     scale: usize,
 ) -> Result<Box<dyn Array>>
 where
-    T: NativeType + AsPrimitive<i128>,
+    T: NativeType + Float + ToPrimitive,
+    f64: AsPrimitive<T>,
 {
     let from = from.as_any().downcast_ref().unwrap();
-    Ok(Box::new(integer_to_decimal::<T>(from, precision, scale)))
+    Ok(Box::new(float_to_decimal::<T>(from, precision, scale)))
```
Comment (houqp) [inline src/compute/cast/primitive_to.rs:242]: this is the fix

**Comment by houqp** [comment]
@Igosuki FYI

**Comment by jorgecarleitao** [comment]
Thanks a lot for the fix!

**Review by alamb — COMMENTED** [review]
🎉

**Comment by Igosuki** [comment]
👍

**Pull request merged**